### PR TITLE
Updatetemplate; Fix model slowness when using threading; Update WW3 for porting and threading issues

### PR DESCRIPTION
**Pull request by DeniseWorthen**
* Updates PR template with added PR checklist
* Fix model slowness when using threading [#258](https://github.com/NOAA-EMC/fv3atm/pull/258)
* Updates WW3 to the latest develop branch, which has updates for threading (see Issue #367). It also has updates for WW3 to be able to be build on gaea, jet and theoretically with gnu compilers although that has not been tested.

Dependent PRs:

FV3-atm: https://github.com/NOAA-EMC/fv3atm/pull/258

RT logs:

- [x] hera.intel
- [x] hera.gnu
- [ ] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3

### CODE REVIEW

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by climbfuji — COMMENTED** [review]
I think we are becoming overly restrictive with the rules around issues. Certainly, there should be one issue for a set of PRs that describes the work, but requiring one for each submodule is unnecessary extra work. And how about small bug fixes that we often combine with other PRs to keep the commit queue shorter? A separate PR for each one-line change?

File: .github/pull_request_template.md
Comment (junwang-noaa) [inline .github/pull_request_template.md]: I also feel maybe this is too strict to have issues in all the subcomponent repositories. E.g. anytime there is a CCPP update,  we need to update fv3 to point to the new CCPP version (also update ufs-weather-model to point to new FV3), since CCPP has an issue already, we usually list the CCPP issue in the FV3 PR (or ufs-weather-model PR), otherwise we may have many FV3/ufs-weather-model issues just for just updating subcomponent version. I think we do need at least one issue for a set of PR, and an issue in subcomponent repository if there are scientific/technical changes in the repository.

**Comment by DeniseWorthen** [comment]
Thanks, I actually thought I was stating the current policy, not adding further restrictions. I myself have found it a bit of a pain to create issues in sub-components and then again in ufs-weather. 

Is the first item (that the PR should be up-to-date w/ all components that are not being changed) OK? 

For the second item, my preference is that the primary issue should be created on the subcomponent page; that one should describe what is being changed. It should have enough detail so that the code managers and reviewers are clear on what/why something is being done. 

On the ufs-weather side, I'm not sure there needs to be an issue at all in a lot of cases---the PR can simply reference the existing issue in the subcomponent. 

I'm happy to make those changes if that is closer to what makes sense to everyone.

File: .github/pull_request_template.md
Comment (junwang-noaa) [inline .github/pull_request_template.md:3]: Do we want to change "the top of all sub-component repositories" to "the versions of all sub-component repositories on the latest ufs-weather-model develop branch"?  This way I think developers will not be confused with the latest version of each sub component repository.

File: .github/pull_request_template.md
Comment (DusanJovic-NOAA) [inline .github/pull_request_template.md]: Please change wcoss-dell-p3 to wcoss_dell_p3.

**Review by junwang-noaa — APPROVED** [review]

File: .github/pull_request_template.md
Comment (MinsukJi-NOAA) [reply .github/pull_request_template.md:3]: One question. Aren't those sub-components which are the subject of this PR need also be up-to-date with the top of their respective repositories? I thought 'up-to-date with the top of ...' means pulling in and merging the top rather than being at the top.

**Review by binli2337 — APPROVED** [review]

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by MinsukJi-NOAA — APPROVED** [review]

**Review by climbfuji — COMMENTED** [review]
Looks good to me, will approve once fv3atm is merged and submodule/.gitmodules is updated.

**Comment by climbfuji** [comment]
Cheyenne is done (both GNU and Intel). Waiting for orion it seems.

**Comment by junwang-noaa** [comment]
What is the status of orion RT? Are some jobs waiting in the queue? Is there any failed test?

**Comment by DeniseWorthen** [comment]
There are 15 jobs in the Q but all completed tests have passed.

**Comment by DeniseWorthen** [comment]
Two tests failed:  fv3_ccpp_gfdlmprad and fv3_ccpp_stretched. The stretched test didn't run at all. I don't see the reason the gfdlmprad failed.

**Review by climbfuji — APPROVED** [review]
Verified that the fv3atm submodule pointer update is correct. Thanks for including the change.

**Pull request merged**
